### DISCUSSION TITLE
Curso: Fundamentos de Asp.Net 6 - Aula: Nomeando um endpoint

### DISCUSSION THREAD

**Discussion by danilobicas-unoeste** [balta-io/forum#569] [Tira Dúvidas] [answered]
Olá Balta.

neste trecho de código você usa o FromServices para pegar o Dbcontext em todos os endpoints de um controller? 
![image](https://github.com/balta-io/forum/assets/12497104/58199e33-8c33-429a-bbbb-c21ca29bb055)

Pergunto: Como ficaria para que o appDBcontext fosse injetado no controller... sem precisar colocar em todos os endpoints dele?
tipo um private readonly AppDBContext _context;? E no construtor? como eu injeto o servico neste _context????

Obrigado

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @danilobicas-unoeste, como vai?

Ficaria assim:

```csharp
public class MeuController : Controller
{
    private readonly AppDbContext _context;
    public MeuController(AppDbContext context) => _context=context;
    
    public IActionResult Get()
    {
         _context.Categories.ToList();
    }

}

```

Caso esteja utilizando .NET 8 + C#12 ou superior, pode usar os Primary Constructors (Mostro isso no curso novo):


```csharp
public class MeuController(AppDbContext context) : Controller
{    
    public IActionResult Get()
    {
         context.Categories.ToList();
    }

}

```
